package.json

```diff
@@ -78,6 +78,7 @@
 	"devDependencies": {
 		"@types/events": "^3.0.3",
 		"@types/jest": "^29.5.12",
+		"@types/node": "20",
```

Comment:
I don't understand. Why do we need types/node dep and why do we need it in devDependencies? Which issue does this exactly fix?

Reply:
Simply put, `jest` depends on `@types/node` type declarations, and it is standard implicit dependencies should be added. (Hoisted `node_modules` by `pnpm` default)

Running `tsc --noEmit` results in errors as follows: 

<details>
<summary>Logs</summary>

```
src/EnhancedEventEmitter.ts:112:2 - error TS2416: Property 'listeners' in type 'EnhancedEventEmitter<E>' is not assignable to the same property in base type 'EventEmitter'.
  Type '<K extends keyof E & string>(eventName: K) => Function[]' is not assignable to type '(type: string | number) => Listener[]'.
    Type 'Function[]' is not assignable to type 'Listener[]'.
      Type 'Function' is not assignable to type 'Listener'.
        Type 'Function' provides no match for the signature '(...args: any[]): void'.

112  listeners<K extends keyof E & string>(eventName: K): Function[] {
     ~~~~~~~~~

src/EnhancedEventEmitter.ts:116:2 - error TS2416: Property 'rawListeners' in type 'EnhancedEventEmitter<E>' is not assignable to the same property in base type 'EventEmitter'.
  Type '<K extends keyof E & string>(eventName: K) => Function[]' is not assignable to type '(type: string | number) => Listener[]'.
    Type 'Function[]' is not assignable to type 'Listener[]'.

116  rawListeners<K extends keyof E & string>(eventName: K): Function[] {
     ~~~~~~~~~~~~

src/test/test.ts:1723:28 - error TS2304: Cannot find name 'global'.

1723  const originalNavigator = global.navigator;
                                ~~~~~~

src/test/test.ts:1734:39 - error TS2304: Cannot find name 'global'.

1734     const originalRTCRtpTransceiver = global.RTCRtpTransceiver;
                                           ~~~~~~

src/test/test.ts:1737:6 - error TS2304: Cannot find name 'global'.

1737      global.RTCRtpTransceiver = class Dummy {
          ~~~~~~

src/test/test.ts:1745:5 - error TS2304: Cannot find name 'global'.

1745     global.RTCRtpTransceiver = originalRTCRtpTransceiver;
         ~~~~~~

src/test/test.ts:1752:2 - error TS2304: Cannot find name 'global'.

1752  global.navigator = originalNavigator;
      ~~~~~~


Found 7 errors in 2 files.

Errors  Files
     2  src/EnhancedEventEmitter.ts:112
     5  src/test/test.ts:1723
```

</details>

In another sense, since `global` is used, then type definition for `global` is required, and runtime bindings are provided by `node`. I reckon it as one of `devDependencies`.